Cargo.toml

```diff
@@ -181,6 +181,7 @@ recursive = "0.1.1"
 regex = "1.12"
 rstest = "0.26.1"
 serde_json = "1"
+sha2 = "^0.10.9"
```

Comment:
Because datafusion-spark now uses sha2 directly, we extract it as a common dependency